front/src/pages/companies/__stories__/Companies.add.stories.tsx

```diff
@@ -0,0 +1,60 @@
+import { expect } from '@storybook/jest';
+import type { Meta } from '@storybook/react';
+import { userEvent, within } from '@storybook/testing-library';
+
+import { AppPath } from '@/types/AppPath';
+import {
+  PageDecorator,
+  type PageDecoratorArgs,
+} from '~/testing/decorators/PageDecorator';
+import { graphqlMocks } from '~/testing/graphqlMocks';
+import { sleep } from '~/testing/sleep';
+
+import { Companies } from '../Companies';
+
+import { Story } from './Companies.stories';
+
+const meta: Meta<PageDecoratorArgs> = {
+  title: 'Pages/Companies/Add',
+  component: Companies,
+  decorators: [PageDecorator],
+  args: { routePath: AppPath.CompaniesPage },
+  parameters: {
+    docs: { story: 'inline', iframeHeight: '500px' },
+    msw: graphqlMocks,
+  },
+};
+
+export default meta;
+
+export const AddNewCompany: Story = {
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement);
+
+    const table = await canvas.findByRole('table');
+
+    const tableBody = await within(table).findByTestId('t-body');
+
+    const rowsBeforeAdd = within(tableBody).getAllByRole('row');
+
+    const addButton = await canvas.findByTestId('add-button');
+
+    userEvent.click(addButton);
+
+    await sleep(200);
+
+    const rowsAfterAdd = within(tableBody).getAllByRole('row');
+
+    const firstRow = rowsAfterAdd[0];
+    const cells = within(firstRow).getAllByRole('cell');
+
+    cells.forEach((cell, index) => {
+      // empty name
+      if (index === 1) {
+        expect(cell.textContent).toBe('');
+      }
+    });
+
+    expect(rowsAfterAdd.length).toBe(rowsBeforeAdd.length + 1);
```

Comment:
Hi @rtpa25, thank you for contributing!

I think you can remove some steps in this test (same for the test in `People.add.stories.tsx`):

- No need to find `table` and `tableBody`, you can just use `canvas.getAllByRole('row')` to get `rowsBeforeAdd`. You'll need to use the `sleep` function before getting the rows, as they don't load immediately. After testing I found that `await sleep(2000)` should be enough. 
- In the same way, you can also retrieve `rowsAfterAdd` with `canvas.getAllByRole('row')`. Then `firstRow` would be `rowsAfterAdd[1]` (0 is the header row). For me `await sleep(200)` was not enough, I had to increase the waiting time to 1000.
- You don't need to use `forEach` on the cells, only testing if `cells[1].textContent` is empty should work.
- Nice to know: there is a `toHaveLength` matcher, so you could write:
```typescript
expect(rowsAfterAdd).toHaveLength(rowsBeforeAdd.length + 1);
```

**Bonus (optional):** it would be nice for the "add" button to be accessible. So instead of relying on a `data-testid`, we could add an `aria-label` prop to `IconButton`, then in `PageAddButton`, set the `aria-label` to "Add". Then we can get the button like this in the test:
```typescript
const addButton = canvas.getByRole('button', { name: 'Add' });
```